sql/drop.go

```diff
@@ -87,3 +88,67 @@ func (p *planner) DropTable(n *parser.DropTable) (planNode, error) {
 	}
 	return &valuesNode{}, nil
 }
+
+// DropDatabase drops a database.
+func (p *planner) DropDatabase(n *parser.DropDatabase) (planNode, error) {
+	if n.Name == "" {
+		return nil, errEmptyDatabaseName
+	}
+
```

Comment:
This is because you're calling p.DropTable on each table, which checks for table-level WRITE privileges.

For now, it's fine to keep it that way and explain it in the DropDatabase comment as well as the tests.